### PR TITLE
Add dev tools, requirements.txt and format/clean up code

### PR DESCRIPTION
**Pull request by jungerm2**
Hello, thanks for this excellent project! It has saved me a lot of time.

I went ahead and added some development tools that will help with future development/on-boarding new devs. They can be easily accessed through `invoke` as such: 

```
> invoke --list
Available tasks:

  clean          Runs all clean sub-tasks
  clean-build    Clean up files from package building
  clean-python   Clean up python file artifacts
  format         Format code
  lint           Run all linting
  lint-flake8    Lint code with flake8
  lint-pylint    Lint code with pylint
```

These help keep the code clean and consistent. The large diff in this PR is mostly due to the automatic formating that tweaked/formatted code such that it complies with the PEP8 standard. No functionality has changed at all. 

I did also remove unnecessary imports. Like that, a user can use this library without downloading `open3d` (which is only used for testing). 

Finally, I added a `requirements.txt` (which only included numpy, as random is in the stdlib) and a `requirements-dev.txt` which contains the required libraries for testing/maintaining the codebase.

### CODE REVIEW

**Comment by leomariga** [comment]
@jungerm2 Thank you very much for you contribution! Indeed this are very important change. I'm going to have a look at this as soon as possible!

**Review by leomariga — APPROVED** [review]
Looks good! Thank you very much @jungerm2 .

**Comment by leomariga** [comment]
I'll also add the `clean` and `format` tasks to the contributing guidelines and create some Github actions to verify the code format. I'll work on this for the next release. I believe this will have much value to pyRANSAC-3d. Thanks again for your contribution! =)

**Comment by jungerm2** [comment]
Thanks for merging it in so quickly! Here's a sample github action that should lint and test that you can use:

```
# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions

name: Python application

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]

jobs:
  build:
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
      matrix:
        python-version: [3.6, 3.7, 3.8, 3.9]

    steps:
    - uses: actions/checkout@v2
    - name: Set up Python ${{ matrix.python-version }}
      uses: actions/setup-python@v2
      with:
        python-version: ${{ matrix.python-version }}
    - name: Install dependencies
      run: |
        python -m pip install --upgrade pip
        pip install .
        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
    - name: Lint with flake8
      run: |
        # stop the build if there are Python syntax errors or undefined names
        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
        invoke lint-flake8
    - name: Test with pytest
      run: |
        invoke test
```

Note: This wouldn't work with the current tests you have implemented as they are not automated tests. But if/when you move on to something like pytest and add an `invoke test` task, the above will work. I highly recommend using pytest. 

Lastly, I wouldn't add an autoformatter to the github actions, just linting (with a few allowed linting warnings as above).

**Comment by leomariga** [comment]
Thanks! Yes, the test folder is not actually a formal testing script, I'll change it to something like "sample code" in a near future. I'm currently working on creating (real) tests to cover all library.

Just so I can understand better, why wouldn't you add the autoformatter on github actions?

I was thinking in checking the pull request to see if the user used the clean, format and lint before commit (just like is done in Open3D). However, we already struggle to receive contributions and PR, and I believe this would create more barriers for the contributors. 

Maybe the best way is: I make this modifications myself always before every release. Then, add this just as a recommendation in the contributing guideline.

**Comment by jungerm2** [comment]
I think you could do either or. It might make sense to add some details about how to lint/format/test in the contributing doc. 

As far as why I wouldn't autoformat in a github action:
1. I'm not sure that you'll be able to commit the code if it changes
2. And even if you could auto commit it, it's probably better to see what changes you commit instead of doing it blindly
3. Sometimes the formatted might change a file in a way you dislike, that the linter will complain about, etc. 

Basically, it's better to just spend the extra effort to do it yourself pre-commit and then have the github action only see if it passes the linter with no warnings. OIf it does, you can roughly assume the formatting is OK.
